src/Microsoft.OData.Client/Serialization/ODataPropertyConverter.cs

```diff
@@ -611,6 +802,33 @@ private bool TryConvertPropertyToResourceOrResourceSet(ClientPropertyAnnotation
             return false;
         }
 
+        /// <summary>
+        /// Tries to convert the given value into an instance of <see cref="ODataItemWrapper"/>.
+        /// </summary>
+        /// <param name="propertyName">Name of the dynamic property.</param>
+        /// <param name="propertyValue">The property value to convert.</param>
+        /// <param name="serverTypeName">The server type name of the entity whose properties are being populated.</param>
+        /// <param name="visitedComplexTypeObjects">Set of instances of complex types encountered in the hierarchy. Used to detect cycles.</param>
+        /// <param name="odataItem">The odata resource or resource set if one was created.</param>
+        /// <returns>Whether or not the value was converted.</returns>
+        private bool TryConvertDynamicPropertyToResourceOrResourceSet(string propertyName, object propertyValue, string serverTypeName, HashSet<object> visitedComplexTypeObjects, out ODataItemWrapper odataItem)
+        {
+            Debug.Assert(!string.IsNullOrEmpty(propertyName), "!string.IsNullOrEmpty(propertyName)");
+            Debug.Assert(propertyValue != null, "propertyValue != null");
+            Debug.Assert(serverTypeName != null, "serverTypeName != null");

```

Comment:
Why do we need to check this only in debug mode?

Reply:
@paulodero Responded to this [here](https://github.com/OData/odata.net/pull/1712#discussion_r412895515)